components/brave_search/browser/brave_search_default_host.cc

```diff
@@ -136,6 +150,11 @@ uint64_t BraveSearchDefaultHost::GetMaxTotalCanAskCount() {
       features::kBraveSearchDefaultAPITotalLimit.Get());
 }
 
+void BraveSearchDefaultHost::SetCanAlwaysSetDefault() {
+  can_always_set_default_ =
```

Comment:
Maybe a comment as to why this is conditional on 2 things - 1. this function getting called (because of a URL parameter) and 2. If search promotions are currently active.
At the moment it is confusing to read the code because it's not obvious why we don't set `can_always_set_default_ = IsSearchPromotionEnabled(prefs_, template_url_service_);` in the constructor.

Reply:
Ok, I added a comment like below.
```
  // We don't apply 24h limit if search promotion url is loaded.
  // When renderer detected that current url is for promotion,
  // renderer requests to remove this limit. In that case,
  // limit is removed when promotion is enabled.
```